aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1425,15 +1425,18 @@ AT_ERROR("inverse: MAGMA library not found in "
   magma_int_t lda = std::max<magma_int_t>(1, n);
   magma_int_t lwork = n * magmaGetriOptimalBlocksize<scalar_t>(n);
 
-  // magmaLu and magmaGetri requires infos tensor to live on CPU
-  infos_lu = infos_lu.to(at::kCPU);
-  infos_getri = infos_getri.to(at::kCPU);
+  // magmaLu and magmaGetri requires info argument to live on CPU
+  // but info_lu and info_getri tensors are on the same device as self
+  magma_int_t info_lu_cpu = 0;
+  magma_int_t info_getri_cpu = 0;
 
   Tensor ipiv = at::empty({lda}, at::kInt);
   Tensor dwork = at::empty({lwork}, self.options());
-  magmaLu<scalar_t>(n, n, self_data, lda, ipiv.data_ptr<magma_int_t>(), infos_lu.data_ptr<magma_int_t>());
+  magmaLu<scalar_t>(n, n, self_data, lda, ipiv.data_ptr<magma_int_t>(), &info_lu_cpu);
   magmaGetri<scalar_t>(
-    n, self_data, lda, ipiv.data_ptr<magma_int_t>(), dwork.data_ptr<scalar_t>(), lwork, infos_getri.data_ptr<magma_int_t>());
+    n, self_data, lda, ipiv.data_ptr<magma_int_t>(), dwork.data_ptr<scalar_t>(), lwork, &info_getri_cpu);
+  info_lu.fill_(info_lu_cpu);
+  info_getri.fill_(info_lu_cpu);
```

Comment:
```suggestion
  info_getri.fill_(info_getri_cpu);
```
And a question about this line and the one before. Does `fill_` work as expected with `magma_int_t`  or is it cast first to an `int64_t` and then used to fill the tensor? If the cast is happening, it'd be better to cast it explicitly with `magma_int_cast`.

Reply:
Thanks for spotting the error! `info_getri` is not exposed to the user-facing API as it doesn't contain any additional useful information on the input. I should remove it, but that's for later.

I think `fill_` converts the input to `c10::Scalar`, which stores the integers as `int64_t`.
`magma_int_cast` only does the `int64_t` -> `magma_int_t` conversion. `magma_int_t` is `int`, so there is an implicit conversion from `int` to `int64_t` in `fill_` and then again to `int` because the type of `info` tensor is `at::kInt` (`int`).

Reply:
That makes sense, thank you for the explanation!